logfire/_internal/cli.py

```diff
@@ -276,7 +302,9 @@ def _write_credentials(project_info: dict[str, Any], data_dir: Path, logfire_api
 def parse_create_new_project(args: argparse.Namespace) -> None:
     """Create a new project."""
     data_dir = Path(args.data_dir)
-    logfire_url = args.logfire_url
+    logfire_url = _get_logfire_url(args.logfire_url, args.region)
+    if logfire_url is None:
+        _, logfire_url = LogfireCredentials._get_token_data()  # type: ignore
```

Comment:
Perhaps `_get_token_data()` should be made "public".

Reply:
`LogfireCredentials.create_new_project` already calls `_get_token_data`, which makes sense, so you shouldn't need this.

Reply:
The issue is I need the URL to then call `_write_credentials()` :confused: :

https://github.com/pydantic/logfire/blob/4a93a7674813c977a30709a3ab44cfea05ab5ad8/logfire/_internal/cli.py#L305-L318

Reply:
Yes I mentioned this in https://github.com/pydantic/logfire/pull/901#issuecomment-2749324773, after the region launch we can revisit

Reply:
yes but what i'm saying is that unlike what you said there this logic is *not* necessary